arachne.c

```diff
@@ -493,51 +493,1343 @@ int main(int argc, char *argv[]) {
           double signal = fluxes[i] / sigma;
           double pval = random_deviate(&seed);
 
-          if (in == 3)
-            out = 3;
-          else if (in == 2) {
-            plvl1 = (prob(max(0, lvl - signal)) - 0.5) / (prob(lvl) - 0.5);
-            if (pval < plvl1)
-              out = 2;
-            else
-              out = 3;
-          } else if (in == 1) {
-            plvl1 =
-                (0.5 - prob(clip(-lvl, 0, lvl - signal))) / (0.5 - prob(-lvl));
-            plvl2 = plvl1 + (prob(clip(-lvl, 0, lvl - signal)) -
-                             prob(max(-signal, -lvl))) /
-                                (0.5 - prob(-lvl));
-            if (pval < plvl1)
-              out = 3;
-            else if (pval < plvl2)
-              out = 2;
-            else
-              out = 1;
-          } else if (in == 0) {
-            plvl1 = (prob(-lvl) - prob(min(-signal + lvl, -lvl))) / prob(-lvl);
-            plvl2 = plvl1 + (prob(min(-signal + lvl, -lvl)) -
-                             prob(min(-signal, -lvl))) /
-                                prob(-lvl);
-            plvl3 = plvl2 + (prob(min(-signal, -lvl)) - prob(-lvl - signal)) /
-                                prob(-lvl);
-            if (pval < plvl1)
-              out = 3;
-            else if (pval < plvl2)
-              out = 2;
-            else if (pval < plvl3)
-              out = 1;
-            else
-              out = 0;
-          }
-          raw[I] = out;
+         /*========================================================================*/
+         /*======================== 8 bit FRB Injection ===========================*/
+          switch (in) {
+          case 0:
+            for (m = 255-in; m > 0; m--) {
+              plvl = (prob(min((in -127)*lvl, (in + m -127)*lvl - signal)) - prob(max((in -128)*lvl, (in + m -128)*lvl - signal)))/
```

Comment:
You have not declared `plvl` anywhere, you need to modify this line:

https://github.com/astrogewgaw/arachne/blob/59b44411d6fe6bca8603a0f67542b034e9e6bc6e/arachne.c#L492

to:

```c
double plvl;
```

Reply:
Oh okay, I will make these changes.